lib/puppet/pops/issues.rb

```diff
@@ -152,680 +154,681 @@ def self.hard_issue(issue_code, *args, &block)
-    "Illegal Resource Type expression, expected result to be a type name, or untitled Resource, got #{actual}"
+    _("Illegal Resource Type expression, expected result to be a type name, or untitled Resource, got %{actual}") % { actual: actual }
   end
 
   DUPLICATE_TITLE = issue :DUPLICATE_TITLE, :title  do
-    "The title '#{title}' has already been used in this resource expression"
+    _("The title '%{title}' has already been used in this resource expression") % { title: title }
   end
 
   DUPLICATE_ATTRIBUTE = issue :DUPLICATE_ATTRIBUE, :attribute  do
-    "The attribute '#{attribute}' has already been set"
+    _("The attribute '%{attribute}' has already been set") % { attribute: attribute }
   end
 
   MISSING_TITLE = hard_issue :MISSING_TITLE do
-    "Missing title. The title expression resulted in undef"
+    _("Missing title. The title expression resulted in undef")
   end
 
   MISSING_TITLE_AT = hard_issue :MISSING_TITLE_AT, :index do
-    "Missing title at index #{index}. The title expression resulted in an undef title"
+    _("Missing title at index %{index}. The title expression resulted in an undef title") % { index: index }
   end
 
   ILLEGAL_TITLE_TYPE_AT = hard_issue :ILLEGAL_TITLE_TYPE_AT, :index, :actual do
-    "Illegal title type at index #{index}. Expected String, got #{actual}"
+    _("Illegal title type at index %{index}. Expected String, got %{actual}") % { index: index, actual: actual }
   end
 
   EMPTY_STRING_TITLE_AT = hard_issue :EMPTY_STRING_TITLE_AT, :index do
-    "Empty string title at #{index}. Title strings must have a length greater than zero."
+    _("Empty string title at %{index}. Title strings must have a length greater than zero.") % { index: index }
   end
 
   UNKNOWN_RESOURCE = issue :UNKNOWN_RESOURCE, :type_name, :title do
-    "Resource not found: #{type_name}['#{title}']"
+    _("Resource not found: %{type_name}['%{title}']") % { type_name: type_name, title: title }
   end
 
   UNKNOWN_RESOURCE_PARAMETER = issue :UNKNOWN_RESOURCE_PARAMETER, :type_name, :title, :param_name do
-    "The resource #{type_name.capitalize}['#{title}'] does not have a parameter called '#{param_name}'"
+    _("The resource %{type_name}['%{title}'] does not have a parameter called '%{param}'") % { type_name: type_name.capitalize, title: title, param: param_name }
   end
 
   DIV_BY_ZERO = hard_issue :DIV_BY_ZERO do
-    "Division by 0"
+    _("Division by 0")
   end
 
   RESULT_IS_INFINITY = hard_issue :RESULT_IS_INFINITY, :operator do
-    "The result of the #{operator} expression is Infinity"
+    _("The result of the %{operator} expression is Infinity") % { operator: operator }
   end
 
   # TODO_HEREDOC
   EMPTY_HEREDOC_SYNTAX_SEGMENT = issue :EMPTY_HEREDOC_SYNTAX_SEGMENT, :syntax do
-    "Heredoc syntax specification has empty segment between '+' : '#{syntax}'"
+    _("Heredoc syntax specification has empty segment between '+' : '%{syntax}'") % { syntax: syntax }
```

Comment:
@Magisus @MosesMendoza  Heredoc is a keyword and should likely be left out of translation. what do you think?